packages/payment/src/services/payment.ts

```diff
@@ -0,0 +1,161 @@
+import { Payment } from "@models"
+import {
+  CaptureDTO,
+  Context,
+  CreateCaptureDTO,
+  CreatePaymentDTO,
+  CreateRefundDTO,
+  DAL,
+  FindConfig,
+  RefundDTO,
+  UpdatePaymentDTO,
+} from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+  ModulesSdkUtils,
+} from "@medusajs/utils"
+
+type InjectedDependencies = {
+  paymentRepository: DAL.RepositoryService
+  captureRepository: DAL.RepositoryService
+  refundRepository: DAL.RepositoryService
+}
+
+export default class PaymentService<
+  TEntity extends Payment = Payment
+> extends ModulesSdkUtils.abstractServiceFactory<
+  InjectedDependencies,
+  {
+    create: CreatePaymentDTO
+    update: UpdatePaymentDTO
+  }
+>(Payment)<TEntity> {
+  protected captureRepository_: DAL.RepositoryService
+  protected refundRepository_: DAL.RepositoryService
+  protected paymentRepository_: DAL.RepositoryService
+
+  constructor(container: InjectedDependencies) {
+    // @ts-ignore
+    super(...arguments)
+
+    this.paymentRepository_ = container.paymentRepository
+    this.captureRepository_ = container.captureRepository
+    this.refundRepository_ = container.refundRepository
+  }
+
+  /**
+   * NOTE TODO - TEMP IMPL. THAT WILL CHANGE
+   */
+  protected transformQueryForAmounts(config: FindConfig<TEntity>) {
+    const select = config?.select
+    let relations = config?.relations
+    const amountsSelect: string[] = []
+
+    if (select?.includes("captured_amount")) {
+      amountsSelect.push("captured_amount")
+      if (!relations?.includes("captures")) {
+        if (!relations) {
+          relations = ["captures"]
+        } else {
+          relations?.push("captures")
+        }
+      }
+    }
+
+    if (select?.includes("refunded_amount")) {
+      amountsSelect.push("refunded_amount")
+      if (!relations?.includes("refunds")) {
+        if (!relations) {
+          relations = ["refunds"]
+        } else {
+          relations?.push("refunds")
+        }
+      }
+    }
+
+    return { select, relations, amountsSelect }
+  }
+
+  /**
+   * NOTE TODO - TEMP IMPL. THAT WILL CHANGE
+   */
+  protected decorateAmounts(payment: TEntity, amountsSelected: string[]) {
+    const amounts = {}
+
+    for (const amountName of amountsSelected) {
+      if (amountName === "captured_amount") {
+        amounts["captured_amount"] = payment.captures.reduce(
+          (sum, capture) => sum + Number(capture.amount),
+          0
+        )
```

Comment:
@carlos-r-l-rodrigues, @adrien2p – I am curious to hear your thoughts on this case. 

Say we have `raw_amount = { value: "100", multiplier: 0.01 }` stored in the database alongside the `amount` as number for a `Capture`. The `raw_amount.value` would be serialized to a BigNumber and `amount` to a number upon querying the captures from the database.

Would you use the BigNumber `raw_amount.value` to compute the `captured_amount` in this scenario instead of `amount`?

My immediate thought was to always use the BigNumber now that we know we have it – what do you think?

Reply:
I would always use the bignumber to not loose any precision.